eclass/cmake-utils.eclass

```diff
@@ -652,25 +652,72 @@ enable_cmake-utils_src_compile() {
 }
 
 _ninjaopts_from_makeopts() {
-	if [[ ${NINJAOPTS+set} == set ]]; then
-		return 0
-	fi
-	local ninjaopts=()
-	set -- ${MAKEOPTS}
-	while (( $# )); do
-		case $1 in
-			-j|-l|-k)
-				ninjaopts+=( $1 $2 )
-				shift 2
+	local makeopts="${1:-"${MAKEOPTS}"}" ninjaopts=()
+	local jobs= keep= load=
+
+	set -- ${makeopts}
+	while (( ${#} )) ; do
+		case "${1}" in
+			-j|--jobs)
+				if [ -n "${2}" ] && [[ ${2} =~ ^[0-9]+$ ]] ; then
```

Comment:
It does not break anything. However, maintaining consistent code style is important too.